### PR TITLE
use shared informer to watch kubernetes resources

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
services endpoints nodes and pods resource watch should use shared informer

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller.go [hunk 1]
```diff
@@ -156,25 +137,18 @@ func (c *Controller) notify(obj interface{}, event model.Event) error {
...
-// createInformer registers handlers for a specific event.
+// createCacheHandler registers handlers for a specific event.
 // Current implementation queues the events in queue.go, and the handler is run with
 // some throttling.
 // Used for Service, Endpoint, Node and Pod.
 // See config/kube for CRD events.
 // See config/ingress for Ingress objects
-func (c *Controller) createInformer(
-	o runtime.Object,
+func (c *Controller) createCacheHandler(
```
Comment (ZackButcher) [inline pilot/pkg/serviceregistry/kube/controller.go]: nit: now that this isn't so long, can you bring this function declaration back onto one line pleaes

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller.go]: make sense

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller.go]: @ZackButcher addressed

**Comment by hzxuzhonghu** [comment]
@ZackButcher addressed and rebased, ptal

**Pull request merged**